core/tx_pool.go

```diff
@@ -1706,9 +1722,14 @@ func (pool *TxPool) truncateQueue() {
 // Note: transactions are not marked as removed in the priced list because re-heaping
 // is always explicitly triggered by SetBaseFee and it would be unnecessary and wasteful
 // to trigger a re-heap is this function
-func (pool *TxPool) demoteUnexecutables() {
+func (pool *TxPool) demoteUnexecutables(affectedAccounts map[common.Address]bool) {
+	log.Info("Demoting unexecutable transactions", "affected", len(affectedAccounts))
 	// Iterate over all accounts and demote any non-executable transactions
 	for addr, list := range pool.pending {
+		if affectedAccounts != nil && !affectedAccounts[addr] {
```

Comment:
It is true that only these accounts might have higher nonce or lower balance.

But it is possible that a transaction has sufficient balance before, but it's balance is not sufficient anymore, because of L1 data fee. Would not processing these accounts here cause any problem?

Reply:
Good point, I missed that. In that case, those txns will linger in the txpool until worker tries to execute them. At that point, worker will encounter an ErrInsufficientFunds error and remove it from the pool manually.

Reply:
but it is fine imo. Since L1DataFee increasing is not in control of any user, this should not be a possible attack vector

Reply:
or could we just randomly trigger the fallback demoteUnexecutables()? e.g. 1000 times with one that loops all accounts.

Reply:
so the `affectedAccounts` purpose is that: decreases the `demoteUnexecutables` traversal interval, is it correct? 

if true, maybe `truncatePending` also need to check it